_pytest/main.py

```diff
@@ -210,7 +210,9 @@ def __getitem__(self, key):
             return self.parent.keywords[key]
 
     def __setitem__(self, key, value):
-        self._markers[key] = value
+        from _pytest.mark import MarkDecorator
+        if not isinstance(self._markers.get(key), MarkDecorator):
```

Comment:
the fix just makes it bad in a different way

 the real problem is, that we track markers in a very horrific manner
its simply the completely wrong datastructure